WorkAndFun/src/main/java/pl/infoshare/announcements/AnnouncementSearchService.java

```diff
@@ -0,0 +1,110 @@
+                if (!announcementsByCategory.isEmpty()) {
+                    for (Announcement announcement : announcementsByCategory) {
+                        showAnnouncement(announcement);
+                    }
+                    chooseAndShowAnnouncementDetails();
+                    System.out.println("Powrót do menu wyszukiwania");
+                } else {
+                    System.out.println("brak ogłoszeń");
+                }
+                searchAnnoucement(offerType);
+                break;
+            case 3:
+                System.out.println("Wyszukaj po opisie");
+                var announcementsByDescription = searchByDescription(announcementList, offerType);
+                if (!announcementsByDescription.isEmpty()) {
+                    for (Announcement announcement : announcementsByDescription) {
+                        showAnnouncement(announcement);
+                    }
+                    chooseAndShowAnnouncementDetails();
+                    System.out.println("Powrót do menu wyszukiwania");
+                } else {
+                    System.out.println("brak ogłoszeń");
+                }
+                searchAnnoucement(offerType);
+                break;
+        }
+    }
+
+    private List<Announcement> searchByDescription(List<Announcement> announcementList, OfferType offerType) {
+        String input = scanner.nextLine();
+        return announcementList.stream().filter(announcement -> announcement.getOfferType().equals(offerType) &
+                announcement.getDescription().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT))).collect(Collectors.toList());
+    }
+
+    private List<Announcement> searchByLocalisation(List<Announcement> announcementList, OfferType offerType) {
+        String input = scanner.nextLine();
+        return announcementList.stream()
+                .filter(announcement -> announcement.getOfferType().equals(offerType) &
+                        (announcement.getVoivodeship().toString().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT))
+                                || announcement.getCity().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT))
+                                || announcement.getCityDistrict().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT)) ||
+                                announcement.getUnit().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT)))).collect(Collectors.toList());
+    }
+
+    private List<Announcement> searchByCategory(List<Announcement> announcementList, OfferType offerType) {
+        String input = scanner.nextLine();
+        return announcementList.stream().filter(announcement -> announcement.getOfferType().equals(offerType) &
+                announcement.getServiceType().toString().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT))).collect(Collectors.toList());
```

Comment:
```suggestion
                announcement.getServiceType().getServiceTypeName().toLowerCase(Locale.ROOT).contains(input.toLowerCase(Locale.ROOT))).collect(Collectors.toList());
```
ważna zmiana! Uzytkownik raczej nie będzie wpisywać "budowa_domu", tylko "budowa domu", a więc niech wyszukuje po "ludskiej" nazwie, a nie nazwie samego enuma

Reply:
I w tym własnie momencie wyszło, że nie sprawdziłes jak działa metoda. Wystarczy że użytkownik wpisze "bud" i ogłoszenie mu wyskoczy, jednak nie zmienia to faktu, że dałem ciała, ponieważ faktycznie powinno szukać po serviceTypeName. Zmienione.
Dzięki za solidne i rzeczowe review ! :)